BrainPortal/app/models/boutiques_portal_task.rb

```diff
@@ -773,6 +788,15 @@ def generate_placeholder_descriptor
 
   private
 
+  # Return remaining file ids for the unique mandatory file descriptor input
+  def ids_for_uniq_mandatory_file(descriptor)
+    used_file_ids   = []
+    descriptor.file_inputs.select{|x| x.optional == true }.map(&:id).each do |input_id|
```

Comment:
We have a method `optional_file_inputs` in the descriptor model.

Reply:
Also, in general in Ruby, if you wrute code that has the overall structure

```ruby
variable = [] # or {}
some_loop.each do |xyz|
  variable << something
end
```

then there is probably a better way to do this. The Ruby iterators can return values. In the current case, you could have written more simply as

```
user_file_ids = descriptor.optional_file_inputs.map do |input|
  Array(self.invoke_params[input.id])
end.flatten
```

Also, last note about your code (this is made obsolete by my suggestion above), you did a ".flatten" in your last statement but in your original code, in your loop, you could have avoided it by using `used_file_ids |= blahblah` instead of `+=`